### ISSUE TITLE
Icon requirements

### ISSUE THREAD

**Issue by afc163** [ant-design/ant-design-icons#71]
This issue is the central backlog for new Ant Design icon requests.

Please add new requests as comments here instead of opening a separate issue. A good request should include:

- Icon name or proposed API name, for example `TelegramOutlined`.
- Use case and expected style: outline, filled, two-tone, or brand logo.
- Reference SVG, official brand guideline, or screenshot when possible.
- License/brand usage notes for third-party logos.

## Requests Consolidated From Standalone Issues

The following open icon-request issues were consolidated here on 2026-06-26.

### Brand And Product Logos

- [ ] AI application logos / AI product logos (#716)
- [ ] Telegram logo (#714)
- [ ] Bluesky logo (#691)
- [ ] Safari browser logo (#626)
- [ ] Discord logo (#334)
- [ ] Stack Overflow logo (#519)
- [ ] Fiverr logo (#578)
- [ ] Facebook logo refresh (#343)
- [ ] Twitter / X logo refresh (#608)

### UI And System Symbols

- [ ] SunMoon theme-toggle icon (#675)
- [ ] Dark mode / light mode icon set (#542)
- [ ] Empty circle, square, rectangle, and triangle icons (#237, #466)
- [ ] Basic missing icons: camera flash, camera flip, borderless close, unchecked circle/square (#495)
- [ ] RTL / LTR direction icons (#433)
- [ ] Open-in-new / external-link icon (#537)
- [ ] Holder icon export / `HolderOutlined` request (#541)
- [ ] Road icon and other Bootstrap migration gaps (#539)
- [ ] Airplane and hotel icons (#643)

### Commerce And Finance

- [ ] INR / rupee icon (#617)

### Date And Calendar

- [ ] Calendar date icons for each day of the month (#319)

### Connectivity And Devices

- [ ] Bluetooth icon (#545)

## Requests Already Mentioned In This Thread

These came from comments in this issue and should be verified against the latest icon set before implementation:

### Brand And Product Logos

- [ ] Docker
- [ ] Kubernetes / k8s
- [ ] Twitch
- [ ] Spotify
- [ ] Pinterest
- [ ] Azure
- [ ] AWS
- [ ] Google Cloud Platform / GCP
- [ ] Bilibili
- [ ] Fediverse
- [ ] TikTok
- [ ] WeCom / WeChat Work
- [ ] X logo
- [ ] OpenAI logo
- [ ] UnionPay
- [ ] Linktree
- [ ] Threads
- [ ] Odysee
- [ ] Reddit logo refresh
- [ ] Browser logos: Firefox, Opera, Safari, Edge

### Development And Technical Symbols

- [ ] Java
- [ ] JavaScript
- [ ] Python
- [ ] Ruby
- [ ] .NET
- [ ] PowerShell
- [ ] Git merge
- [ ] Commit

### File, Media, And Communication

- [ ] FileAudio
- [ ] FileVideo
- [ ] Muted video / disabled camera
- [ ] No sound
- [ ] Media player stop
- [ ] SMS / email
- [ ] Send
- [ ] Speaker

### UI And General Symbols

- [ ] Tree
- [ ] Moon / theme icon
- [ ] Signature / pen
- [ ] Truck / logistics
- [ ] Chair / sitting area
- [ ] Product / box
- [ ] External link
- [ ] Anchor
- [ ] Alarm
- [ ] Add alarm
- [ ] Two-person user group without plus
- [ ] Eyedropper / color picker
- [ ] Laptop plus mobile / devices
- [ ] Ice / cold
- [ ] Reverse refresh

## Notes

- Brand-logo requests need extra care around official shapes, naming, and licensing.
- Some older requests may already exist in the latest packages. Please verify before starting implementation.
- Standalone icon-request issues may be closed after they are linked here so this issue remains the single tracking place.

**Comment by afc163** [comment]
- [ ] docker https://github.com/ant-design/ant-design/issues/17301
- [ ] k8s https://github.com/ant-design/ant-design/issues/17301

**Comment by pdeva** [comment]
- [ ] java
- [ ] javascript
- [ ] python
- [ ] ruby
- [ ] dotnet

**Comment by wzhudev** [comment]
- [ ] send #36

**Comment by JStuve** [comment]
Twitch Icon please!

**Comment by skix123** [comment]
Hi, I'd appreciate these new icons:
- a tree
- a speaker
- Spotify logo

Thanks in advance!

**Comment by afc163** [comment]
- [ ] UPI：https://github.com/ant-design/ant-design/issues/20653

**Comment by vagusX** [comment]
ping @AntDesigners

**Comment by afc163** [comment]
这样 ping 没什么用。

**Comment by flashblaze** [comment]
@afc163 I have tried to contribute Gumroad icon #219 can you tell me whether I've met the requirements and let me know regarding the changes.

Thanks.

**Comment by hzgotb** [comment]
eye 和 eye-invisible 在 文档上没出现。

**Comment by Yumcoder-dev** [comment]
moon icon for dark mode (such as telegrama and twitter night theme)

**Comment by AlonGvili** [comment]
if you can create PowerShell icon it will be great.
and maybe make all the theme icon style the same, for example in the TwoTone style there are less icons then outline

**Comment by afc163** [comment]
- [ ] Pinterest https://github.com/ant-design/ant-design/issues/23458

**Comment by afc163** [comment]
- [ ] FileAudio https://github.com/ant-design/ant-design/issues/23838
- [ ] FileVideo https://github.com/ant-design/ant-design/issues/23838

**Comment by LucaFilitz** [comment]
Why do you close the issues when the icons aren't implemented?

**Comment by afc163** [comment]
@LucaFilitz They all were merged into this issue and you can trace here.

**Comment by fredericosilva** [comment]
Hello, I'd appreciate those,

- [ ] azure - Microsoft Azure
- [ ] aws - Amazon Web Services
- [ ] gcp - Google Cloud Platform

Thanks in advance!

**Comment by LucaFilitz** [comment]
Is there a plan when you'll start implementing the icons? Can I help by a PR? For example the docker and kubernetes Icons aren't in and the comment is almost a year old.

**Comment by ChuckJonas** [comment]
Something to indicate a "signature" required.

Maybe just a "pen"?

**Comment by BrandonKlotz** [comment]
Git Merge icon similar to: https://primer.style/octicons/git-merge-16

@afc163 I am happy to help contribute new Icons to the specification that Ant.Design has listed in: https://github.com/ant-design/ant-design-icons/blob/master/packages/icons-svg/docs/ContributionGuide.md

Is there a separate guide to adding to the Ant.Design site or is it okay to focus on adding to the React Library?

**Comment by afc163** [comment]
- [ ] bilibili https://github.com/ant-design/ant-design/issues/28258

**Comment by jeggy** [comment]
Hi, I would love to get a External link icon (Example https://image.flaticon.com/icons/svg/25/25284.svg)

This has been requested before here: https://github.com/ant-design/ant-design/issues/18841

**Comment by afc163** [comment]
- [ ] muted video https://github.com/ant-design/ant-design/issues/31033

**Comment by sanketshevkar** [comment]
> * [ ]  muted video [ant-design/ant-design#31033](https://github.com/ant-design/ant-design/issues/31033)

@afc163 May I add a SVG and create a PR on the [ant-design-icons ](https://github.com/ant-design/ant-design-icons) repo?

**Comment by afc163** [comment]
- [ ] commit https://github.com/ant-design/ant-design-icons/issues/487

**Comment by balboFK** [comment]
Hi everyone. 
I would like to request the following icons: 

- [ ] a "truck" icon for representing logistics 
- [ ] a chair icon for representing a sitting area
- [ ] a product icon (could be a box of some sort)

Thanks in advance!

**Comment by gabek** [comment]
Hello!  I'd love to see the inclusion of the Fediverse logo as shown here: https://en.wikipedia.org/wiki/Fediverse#/media/File:Fediverse_logo_proposal.svg

**Comment by jadzeidan** [comment]
> Hi, I would love to get a External link icon (Example https://image.flaticon.com/icons/svg/25/25284.svg)
> 
> This has been requested before here: [ant-design/ant-design#18841](https://github.com/ant-design/ant-design/issues/18841)

Yes, would love to have an external link icon, something like this: 

![image](https://user-images.githubusercontent.com/42831937/129590563-7582d460-8705-4b17-ac33-ca2214198503.png)

**Comment by alextrastero** [comment]
Tiktok

```
<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 2859 3333" shape-rendering="geometricPrecision" text-rendering="geometricPrecision" image-rendering="optimizeQuality" fill-rule="evenodd" clip-rule="evenodd"><path d="M2081 0c55 473 319 755 778 785v532c-266 26-499-61-770-225v995c0 1264-1378 1659-1932 753-356-583-138-1606 1004-1647v561c-87 14-180 36-265 65-254 86-398 247-358 531 77 544 1075 705 992-358V1h551z"/></svg>
```

**Comment by FernandoBatoni** [comment]
Moon and Sun Icon for theme changing

**Comment by season886** [comment]
wxwork(企业微信)

**Comment by Zzm0809** [comment]
How to customize the icon in your own project and can be referenced in routs.ts

**Comment by GunjanRajTiwari** [comment]
- [ ] No Sound Icon
- [ ] Media Player stop icon

**Comment by afc163** [comment]
- [ ] discord icon https://github.com/ant-design/ant-design/issues/43844

**Comment by RusseII** [comment]
Commit Icon

**Comment by abomblabrom** [comment]
X for the new twitter 
![Screenshot 2023-12-05 at 3 46 55 PM](https://github.com/ant-design/ant-design-icons/assets/1083775/ab8f75a0-df9a-4722-a5c6-32d69aafc8be)

**Comment by vagusX** [comment]
openai logo please

<img src="https://github.com/ant-design/ant-design-icons/assets/6828924/cd8ff5aa-9f13-4104-bee2-c1d64118155b" width="300" />

**Comment by afc163** [comment]
@jadzeidan There it is.

<img width="172" alt="图片" src="https://github.com/ant-design/ant-design-icons/assets/507615/8db57773-25bc-4551-9daa-9090ba87311f">

**Comment by HappyAxin** [comment]
### What problem does this feature solve?
Please add more icons to the icon library！！！
It would be better if it can be compatible with the icons under [Official Icon Library](https://www.iconfont.cn/collections/index?spm=a313x.collections_index.i1.da2e3581b.5ed03a8153MApZ&type=1).

### What does the proposed API look like?
<IconName color={"#HEX"} size={Number} animation={boolean} style={{}}/> is better

**Comment by YarinDay** [comment]
> * [ ]  muted video [Addition of disabled video svg icon ant-design#31033](https://github.com/ant-design/ant-design/issues/31033)

Hey there! @afc163 , didn't see any change in the Icons, still there is not Disable Camera like the Mute Audio icon, Am I right?

**Comment by yimiaodaren** [comment]
- [ ] unionpay ? ok? <img width="176" alt="image" src="https://github.com/ant-design/ant-design-icons/assets/5618443/0c069a55-3b9e-477d-909b-f40cf8839d01">

**Comment by ItsMalma** [comment]
<img width="282" alt="Screenshot 2023-12-28 084603" src="https://github.com/ant-design/ant-design-icons/assets/113444590/d6976c64-f150-4970-88d8-e794a5c7783e">

Anchor

**Comment by anhthang** [comment]
Linktree icon please

<img src="https://github.com/ant-design/ant-design-icons/assets/10458061/b10532f5-79f2-4cc3-b24e-41bdb896c82a"  width="200px">

**Comment by anhthang** [comment]
Threads logo

<img src="https://github.com/ant-design/ant-design-icons/assets/10458061/555d02d6-2f5d-40bd-ba75-6245e3bfc40c"  width="200px">

**Comment by WeebNetsu** [comment]
I would love if you could add an icon for Odysee ( https://odysee.com ), thanks!

**Comment by afc163** [comment]
- [ ] RTL/LTR https://github.com/ant-design/ant-design/issues/41338

**Comment by Petersoj** [comment]
New Reddit logo:
![reddit](https://github.com/ant-design/ant-design-icons/assets/14127217/22325122-7450-4dcc-8a31-b19ff6e2d00e)

**Comment by jamoreal** [comment]
sun and moon

**Comment by yoshinotcs** [comment]
Alarm ![alarm](https://github.com/user-attachments/assets/9e323213-4630-46ee-89a3-218116b217b8)
AddAlarm ![add_alarm](https://github.com/user-attachments/assets/39108703-e880-4f56-86ca-2b808f38b686)

**Comment by gregor-mueller** [comment]
![image](https://github.com/user-attachments/assets/4e148199-c0e7-437a-8d9b-a81d92ce262c)
`<UsergroupOutlined />`

2 users together, like this without the `+` (and with the front users missing shoulder/arm 😉 ):  
![image](https://github.com/user-attachments/assets/19196e98-b5b1-49b0-9cab-d841b8432f27)

**Comment by PaulleDemon** [comment]
Can we have an eyedropper / color picker icon? Bootstrap icons has something like this

![eye dropper](https://github.com/user-attachments/assets/238a7e11-6493-4ccf-b974-e369762b57bb)

**Comment by ianzone** [comment]
SunMoon https://github.com/ant-design/ant-design-icons/issues/675

**Comment by Lyumih** [comment]
DevicesOutlined - for show laptop / mobile apps. 

![Image](https://github.com/user-attachments/assets/547d168a-89b0-47e0-874d-9da866dfbe00)


Similar icon `<LaptopOutlined /> copied 🎉` in Ant.

![Image](https://github.com/user-attachments/assets/b0fea7f4-93c4-41a1-ae11-b61a03b1f20f)

Analog in MUI. (Transfer from MUI to ANT library)

https://mui.com/material-ui/material-icons/?query=Device&selected=Phonelink

![Image](https://github.com/user-attachments/assets/a233872b-64ea-4aa2-a9fc-cf612007a2f3)

Example SVG for icon
```
<svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M12 20.8125H2.29655C1.77623 20.8125 1.4153 20.2969 1.59108 19.807L3.00905 14.8125V3.9375C3.00905 3.52266 3.3442 3.1875 3.75905 3.1875H20.259C20.6739 3.1875 21.009 3.52266 21.009 3.9375V7H19.3215V4.875H4.69655V14.1328H12V15.6328H4.53014L3.53873 19.125H8.70202L9.14967 17.093C9.16842 17.0062 9.24342 16.9453 9.33249 16.9453H12V18.2578H10.2372L10.0473 19.125H12V20.8125Z" fill="#989EAB"/>
<path d="M17.5625 19.5625C17.5625 19.8111 17.6613 20.0496 17.8371 20.2254C18.0129 20.4012 18.2514 20.5 18.5 20.5C18.7486 20.5 18.9871 20.4012 19.1629 20.2254C19.3387 20.0496 19.4375 19.8111 19.4375 19.5625C19.4375 19.3139 19.3387 19.0754 19.1629 18.8996C18.9871 18.7238 18.7486 18.625 18.5 18.625C18.2514 18.625 18.0129 18.7238 17.8371 18.8996C17.6613 19.0754 17.5625 19.3139 17.5625 19.5625Z" fill="#989EAB"/>
<path fill-rule="evenodd" clip-rule="evenodd" d="M22 9H15C14.4477 9 14 9.44771 14 10V22C14 22.5523 14.4477 23 15 23H22C22.5523 23 23 22.5523 23 22V10C23 9.44772 22.5523 9 22 9ZM15.7 10.7V21.3H21.3V10.7H15.7Z" fill="#989EAB"/>
</svg>
```

**Comment by rakleed** [comment]
It would be nice to have a Telegram logo.

<img alt="" src="https://github.com/user-attachments/assets/d95fd087-ec13-4d8a-bf18-8a638e09aadf" width="150" />

**Comment by cw1997** [comment]
Browsers:

- [ ] Firefox
- [ ] Opera
- [ ] Safari
- [ ] Edge

**Comment by Lyumih** [comment]
Please add icon for SMS / Email . 

For example: 

![Image](https://github.com/user-attachments/assets/55ec0131-89a0-4d78-b53b-82b311d85dc2)

![Image](https://github.com/user-attachments/assets/4b8da7d5-86a6-4932-8a50-e45fae3e05fb)

![Image](https://github.com/user-attachments/assets/1895f4e7-35bc-460d-8c1c-52d892e89568)

![Image](https://github.com/user-attachments/assets/02a34a24-d1f6-4dbf-a7be-70d1386d4264)

**Comment by balboFK** [comment]
I see that we have an Icon for "fire" now, would be nice to have an icon to "ice" aswell. To represent something as "hot" or "cold". Or literal fire and ice.

**Comment by Lyumih** [comment]
Please add RefreshIcon. It's revers to <ReloadOutlined />

<img width="490" height="277" alt="Image" src="https://github.com/user-attachments/assets/81729537-51ee-47af-863e-a64bf0c75893" />

Missing this
<img width="790" height="278" alt="Image" src="https://github.com/user-attachments/assets/fa91841e-b59d-4a3d-b904-18340d1a73ae" />

**Comment by anotheri** [comment]
telegram, please
